specs/makeStyles.md

```diff
@@ -0,0 +1,290 @@
+# makeStyles RFC
+
+## Background
+
+`makeStyles` is a way how CSS is applied to core components and their variants as well as a utility used to implement style overrides for those components.
+
+## Prior Art
+
+This section summarises previous approaches and their pros and cons.
+
+### v0 approach
+
+`@fluentui/react-northstar` (aka v0 aka Stardust) uses CSS-in-JS. [FelaJS](https://github.com/robinweser/fela) CSS-in-JS library is used.
+
+#### Fela uses atomic CSS classes
+
+- one CSS property = one CSS class
+- all selectors have one class specificity
+- the same classes are reused anywhere in the application = smaller DOM
+
+#### CSS classes are added to DOM when used for the first time
+
+- only the classes that are used are added to DOM
+- that means those are added in non-deterministic order
+
+#### Two CSS classes setting the same property cannot be applied to the same element
+
+If there are two classes setting the same CSS property (`.r {color: red}` and `.b {color: blue}` those can never be applied to the same element.
+As both classes have the same one class specificity, the rule defined later in the DOM will win. As the insertion order is non-deterministic, the resulting style would be non-deterministic as well.
+As a consequence, there is a couple of restrictions:
+
+- Shorthand css properties (`margin`) must be expanded before merging (to `margin-top`, `margin-right`, `margin-bottom`, `margin-left`).
+- To apply any overrides, it is not possible to just concatenate classes. Style objects must be always deep merged and then the classes can be computed.
```

Comment:
I think that we are missing an explanation why deep merge is required (because of nested and pseudo selectors)